### PR TITLE
Add examples to the `JsonParser` documentation

### LINKED ISSUE #1972: No example in JsonParser/JsonElement specifications
I never used Gson, but it seems `JsonParser` and `JsonElement` would best mimic how `org.json` works. It would be clearer if the specifications for at least one of these could include an example demonstrating their use though.

An example of JsonParser is available at https://github.com/google/gson/blob/master/extras/src/main/java/com/google/gson/extras/examples/rawcollections/RawCollectionsExample.java

### PR DESCRIPTION
**Pull request by 33nya**
### Purpose

Closes #1972.
* Add examples to the `JsonParser` specifications to demonstrate the usage of the three parse functions.

### Description

* Add examples to the `parseString​(String json)`, `parseReader​(Reader reader)`, and `parseReader​(JsonReader reader)` functions in `JsonParser` class.

### Checklist

- [x] New code follows the [Google Java Style Guide](https://google.github.io/styleguide/javaguide.html)\
  This is automatically checked by `mvn verify`, but can also be checked on its own using `mvn spotless:check`.\
  Style violations can be fixed using `mvn spotless:apply`; this can be done in a separate commit to verify that it did not cause undesired changes.
- [ ] If necessary, new public API validates arguments, for example rejects `null`
- [ ] New public API has Javadoc
    - [ ] Javadoc uses `@since $next-version$`  
      (`$next-version$` is a special placeholder which is automatically replaced during release)
- [ ] If necessary, new unit tests have been added  
  - [ ] Assertions in unit tests use [Truth](https://truth.dev/), see existing tests
  - [ ] No JUnit 3 features are used (such as extending class `TestCase`)
  - [ ] If this pull request fixes a bug, a new test was added for a situation which failed previously and is now fixed
- [ ] `mvn clean verify javadoc:jar` passes without errors

### CODE REVIEW

File: gson/src/main/java/com/google/gson/JsonParser.java
Comment (Marcono1234) [inline gson/src/main/java/com/google/gson/JsonParser.java:41]: To be bit more operating system agnostic, and to avoid "...json.json", what about using just `my-data.json` or similar as name?
```suggestion
 * try (Reader reader = new FileReader("my-data.json")) {
```

Maybe it should also explicitly specify the charset; otherwise users might encounter corruption of their data:
```suggestion
 * try (Reader reader = new FileReader("my-data.json", StandardCharsets.UTF_8)) {
```

File: gson/src/main/java/com/google/gson/JsonParser.java
Comment (Marcono1234) [inline gson/src/main/java/com/google/gson/JsonParser.java:46]: Maybe we should omit this `catch (IOException e)`; it is not that relevant for this example and users probably know how to handle it appropriately for their application.

File: gson/src/main/java/com/google/gson/JsonParser.java
Comment (Marcono1234) [inline gson/src/main/java/com/google/gson/JsonParser.java:65]: Maybe we should omit this `catch (IOException e)`; it is not that relevant for this example and users probably know how to handle it appropriately for their application.

File: gson/src/main/java/com/google/gson/JsonParser.java
Comment (Marcono1234) [inline gson/src/main/java/com/google/gson/JsonParser.java:43]: Maybe to match the code style of this project and of the other code examples in the Javadoc, might be good to use only 2 spaces instead of 4 spaces for indentation.

(applies to the code snippets below as well)

File: gson/src/main/java/com/google/gson/JsonParser.java
Comment (Marcono1234) [inline gson/src/main/java/com/google/gson/JsonParser.java:57]: Maybe switch this to avoid a "Yoda condition", since `nextName()` should be non-null:
```suggestion
 *         if (fieldName.equals("skipKey")) {
```

File: gson/src/main/java/com/google/gson/JsonParser.java
Comment (Marcono1234) [inline gson/src/main/java/com/google/gson/JsonParser.java:29]: Maybe it would also be useful to mention here that the JSON data is parsed in lenient mode (though each of the methods already says this as well).
```suggestion
 * 
 * <p>The JSON data is parsed in {@linkplain JsonReader#setStrictness(Strictness) lenient mode}.
 *
```

File: gson/src/main/java/com/google/gson/JsonParser.java
Comment (Marcono1234) [inline gson/src/main/java/com/google/gson/JsonParser.java:61]: Maybe this should use a `JsonObject` as well; otherwise it would have been easier to just call `JsonReader.nextString()` here.

**Review by Marcono1234 — COMMENTED** [review]
Thanks a lot for these changes!

I have added a few suggestions as comments. Could you please run `mvn spotless:apply` to fix the formatting issues, to make the CI build succeed? Don't worry too much about having separate commits or the commit messages. The commits will be squashed on merge anyway.

If you want you can also remove the `#1972` reference from the title of the pull request, it is not needed (and might cause confusion after merge, when GitHub also adds the PR number, e.g. `... (#1972) (#2647)`). You can edit the pull request title by clicking the "Edit" button at the top right of the GitHub UI.

**Comment by 33nya** [comment]
@Marcono1234 Thank you for your suggestion! It is very clear and helpful. I adjusted the content of the documentation. Please let me know if there is anything that needs to be improved.

File: gson/src/main/java/com/google/gson/JsonParser.java
Comment (Marcono1234) [inline gson/src/main/java/com/google/gson/JsonParser.java:53]: Really minor, but maybe for pretty-printing consistency and to make it a bit easier to read, there should be a space after the `,`:
```suggestion
 * String json = "{\"skipObj\": {\"skipKey\": \"skipValue\"}, \"obj\": {\"key\": \"value\"}}";
```

**Review by Marcono1234 — COMMENTED** [review]
Thanks a lot for the changes!

**Review by eamonnmcmanus — APPROVED** [review]

**Comment by eamonnmcmanus** [comment]
> @eamonnmcmanus, or do you want to merge the changes anyway regardless of that change? If necessary it can also be done as part of another documentation related pull request.

OK, I'll merge as-is, and if you or the original author want you can send a follow-up PR.

**Comment by 33nya** [comment]
@Marcono1234 Sorry for causing any inconvenience. I misunderstood and wasn't sure if I could modify the PR after it had been approved, so I didn't make any changes. If it's convenient, could you assist with the adjustments in your current documentation PR? Otherwise, I will make the changes when I modify another documentation PR. Thank you!

**Comment by Marcono1234** [comment]
@33nya, no problem. I have included that small change in #2654 now, but not sure if / when that will be merged.
If you are planning to make some other (documentation) changes anyway, feel free to include it there as well. If not, then it is also not a problem I think; it is probably not worth it to have a separate pull request just for this small change.

**Comment by 33nya** [comment]
@Marcono1234 Alright, I understand. Thank you for your assistance!

**Pull request merged**
